contracts/aggregator/fantom/flashloan/main.sol

```diff
@@ -0,0 +1,65 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+import './helper.sol';
+import '../../common/main.sol';
+
+/**
+ * @title Flashloan.
+ * @dev Flashloan aggregator for Fantom.
+ */
+
+contract FlashAggregatorFantom is FlashAggregator {
+    event LogFlashloan(address indexed account, uint256 indexed route, address[] tokens, uint256[] amounts);
+
+    /**
+     * @dev Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @notice Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @param _tokens token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _route route for flashloan.
+     * @param _data extra data passed.
+     */
+    function flashLoan(
+        address[] memory _tokens,
+        uint256[] memory _amounts,
+        uint256 _route,
+        bytes calldata _data,
+        bytes calldata // kept for future use by instadapp. Currently not used anywhere.
+    ) external {
+        require(_tokens.length == _amounts.length, 'array-lengths-not-same');
+        require(routeStatus[_route] == true, 'route-disabled');
+
+        (_tokens, _amounts) = bubbleSort(_tokens, _amounts);
+        validateTokens(_tokens);
+
+        implToCall = routeToImpl[_route];
+
+        Address.functionDelegateCall(implToCall, msg.data, 'call-to-impl-failed');
+
+        delete implToCall;
+        emit LogFlashloan(msg.sender, _route, _tokens, _amounts);
+    }
+}
+
+contract InstaFlashAggregatorFantom is FlashAggregatorFantom {
+    function initialize(address owner_, address aave_, address fla_) public {
+        require(status == 0, 'cannot-call-again');
+        require(ownerStatus == 0, 'only-once');
+        owner = owner_;
+        ownerStatus = 1;
+        status = 1;
+        routeToImpl[9] = aave_;
+        routeStatus[9] = true;
+        routes.push(9);
+        routeToImpl[10] = fla_;
+        routeStatus[10] = true;
+        routes.push(10);
+    }
+
+    // Fallback function
+    fallback(bytes calldata input) external payable returns (bytes memory output) {
```

Comment:
Yes, a fallback function can either have no params, in that case, it would not return anything, or it can take bytes calldata as an input, and that would return a bytes output. This works fine after testing.

Reply:
I read it cannot have these in the resources below
https://www.tutorialspoint.com/solidity/solidity_fallback_function.htm
https://www.geeksforgeeks.org/solidity-fall-back-function/

Reply:
Might have come in later versions?

Reply:
Yes, In geeksforgeeks they're using v4.0, might be it wasn't there in that version.
It's there in the solidity v8.0 docs- https://docs.soliditylang.org/en/v0.8.0/contracts.html?highlight=fallback%20(bytes%20calldata%20input)#fallback-function

Reply:
Gotcha thanks!